x-pack/solutions/search/plugins/search_playground/public/utils/user_query.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import deepEqual from 'fast-deep-equal';
+import { i18n } from '@kbn/i18n';
+import type { ChatForm, ChatFormFields, UserQueryValidations } from '../types';
+
+export const validateUserElasticSearchQuery = (
+  userQuery: string | undefined,
+  elasticsearchQuery: { retriever: any }
+): UserQueryValidations => {
+  if (!userQuery) {
+    return { isValid: false, isUserCustomized: false };
+  }
+  let userQueryErrors: string[] | undefined;
+  if (!userQuery.includes('{query}')) {
+    userQueryErrors = [
+      i18n.translate('xpack.searchPlayground.userQuery.errors.queryPlaceholder', {
+        defaultMessage: 'User query must contain "{query}" placeholder',
+        values: { query: '{query}' },
```

Comment:
i18n wants to replace the `{query}` in the default message so we give it `{query}` :/ is there a better way to handle this I'm unaware of?